dependencies/src/main/java/com/google/cloud/tools/opensource/dependencies/DependencyPath.java

```diff
@@ -25,52 +25,88 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.stream.Collectors;
+import javax.annotation.Nullable;
 import org.eclipse.aether.artifact.Artifact;
 import org.eclipse.aether.graph.Dependency;
 
 /**
- * Path from the root to a node in a dependency tree, where a node is a dependency.
+ * Path from the root to a node in a dependency tree, where the root node is a Maven artifact and
+ * non-root nodes are dependencies.
+ *
+ * <p>The root node is null for the dependency trees generated for multiple artifacts by {@link
+ * DependencyGraphBuilder#buildFullDependencyGraph(List)}; otherwise the root node is not null.
  */
 public final class DependencyPath {
 
-  private List<Dependency> path = new ArrayList<>();
+  // The root of the dependency path. The project root is not a dependency.
+  private final Artifact root;
+  // Path without the root
+  private final List<Dependency> path = new ArrayList<>();
+
+  public DependencyPath(@Nullable Artifact root) {
+    this.root = root;
+  }
 
   @VisibleForTesting
-  public void add(Dependency dependency) {
-    path.add(dependency);
+  public DependencyPath appended(Dependency dependency) {
```

Comment:
"appended" is strangely phrased. Usually this would be a present tense verb.

Reply:
Renamed to `append`.